_posts/2021-01-22-x-ways-to-write-meaningful-commit-messages.md

```diff
@@ -0,0 +1,140 @@
+
+Header: Mandatory - Up to 50 characters (as Git suggests)
+
+> (type) (scope): (short summary)
+
+Description: Mandatory (except for docs) - at least 20 characters
+> (body)
+
+Footer: Optional
+> (footer)
+
+The header consists of a type and a summary part. Some add scope in between. But since it's optional, we only discuss it briefly.
+
+#### Type:
+The type of commit message is that change made for a particular problem. For example, if you've fixed a bug or add a feature, or maybe changed something related to the docs, the type would be Fix, Add, or Doc. It could be whatever you and your teammates decide to be as long as it's understandable and consistent. It's better to be capital.
+
+#### Scope:
+It's the package or module that is affected by the change. As mentioned before, it's optional.
+
+#### Summary:
+As Angular suggests: "It should be present tense. Not capitalized. No period in the end.", and **imperative like the type**.
+As [Chris Beams](https://chris.beams.io/posts/git-commit/) mentions here, the summary should always be able to complete the following sentence: 
+
+If applied, this commit will ...
+
+**If applied, this commit will** 
+add authorization for document access
+
+Bellow, part "Fix" is the type, and the sentence after ":" is the summary.
+> **Right**: Fix: add authorization for document access
+> 
+> **Wrong**: Fix: Add authorization for document access (Capitalized)
+>
+> **Wrong**: Fix: added authorization for document access (Not present tense)
+>
+> **Wrong**: Fix: add authorization for document access. (Used period in the end)
+
+#### Body:
+The format should be just like the summary, but the content goal is different. I should explain the motivation for the change.
+In other words, it should be an imperative sentence explaining why you're changing the code, **compared to what it was before**.
+
+#### Footer: 
+You can mention the related task URL or the issue number in this section.
+
+### Consistency in the format:
+All the rules above are beneficial only if you keep doing it in all your commits. If the structure changes in each commit,
+the Git log would be unstructured and unreadable over time. Which misses the whole point of making these rules.
+
+## Examples:
+#### Commit message 1:
+> 
+> Doc: fix typo in tutorial
+>
+> change colour to color in the introduction
+>
+> Document-130
+>
+
+
+#### Commit message 2:
+>
+>Test: add negative test for payment
+>
+>add test scenario of user with zero credit
+>
+>Payment-145 (or you can use the URL)
+
+#### Commit message 3:
+You can even mix the type and description and wrtie the message like this:
+>
+>Fix typo in tutorial
+>
+>change colour to color in the introduction
+>
+>Document-130
+
+## Conclusion:
+
+A great format for writing commit messages can be different in each team. The most important aspect is to keep it simple, readable, and consistent.
+
+## References:
+
+> [Angular](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit) ,
```

Comment:
Instead of listing the references in the end, please add links to them in the text when you talk about it. Few people click on links in the bottom of a page, but more will do so if the link is embedded in the text.